episodes/14-plot-raster-data.Rmd

```diff
@@ -0,0 +1,291 @@
+
+levels(DSM_TUD_df$fct_elevation_cb)
+```
+
+::: callout
+
+# Data tip
+
+Note that 4 break values will result in 3 bins of data.
+
+The bin intervals are shown using `(` to mean exclusive and `]` to mean inclusive. For example: (0, 10] means “from 0 through 10”.
+
+:::
+
+And now we can plot our bar plot again, using the new groups:
+```{r}
+ggplot() +
+  geom_bar(data = DSM_TUD_df, aes(fct_elevation_cb))
+```
+
+And we can get the count of values in each group in the same way we did before:
+```{r}
+DSM_TUD_df |> 
+  count(fct_elevation_cb)
+```
+
+We can use those groups to plot our raster data, with each group being a different color:
+```{r}
+ggplot() +
+  geom_raster(data = DSM_TUD_df , aes(x = x, y = y, fill = fct_elevation_cb)) + 
+  coord_quickmap()
+```
+The plot above uses the default colors inside ggplot for raster objects. We can specify our own colors to make the plot look a little nicer. R has a built in set of colors for plotting terrain, which are built in to the `terrain.colors()` function. Since we have three bins, we want to create a 3-color palette:
+
+```{r}
+terrain.colors(3)
+```
+
+The `terrain.colors()` function returns hex colors - each of these character strings represents a color. To use these in our map, we pass them across using the `scale_fill_manual()` function.
+```{r}
+ggplot() +
+ geom_raster(data = DSM_TUD_df , aes(x = x, y = y, fill = fct_elevation_cb)) + 
+    scale_fill_manual(values = terrain.colors(3)) + 
+    coord_quickmap()
+```
+
+## More Plot Formatting
+
+If we need to create multiple plots using the same color palette, we can create an R object (`my_col`) for the set of colors that we want to use. We can then quickly change the palette across all plots by modifying the `my_col` object, rather than each individual plot.
+
+We can label the x and y axes of our plot too using `xlab` and `ylab`. We can also give the legend a more meaningful title by passing a value to the `name` argument of the `scale_fill_manual()` function.
+```{r}
+my_col <- terrain.colors(3)
+
+ggplot() +
+ geom_raster(data = DSM_TUD_df , aes(x = x, y = y,
+                                      fill = fct_elevation_cb)) + 
+    scale_fill_manual(values = my_col, name = "Elevation") + 
+    coord_quickmap()
+```
+Or we can also turn off the labels of both axes by passing `element_blank()` to the relevant part of the `theme()` function.
+```{r}
+ggplot() +
+ geom_raster(data = DSM_TUD_df , aes(x = x, y = y,
+                                      fill = fct_elevation_cb)) + 
+    scale_fill_manual(values = my_col, name = "Elevation") +
+    theme(axis.title = element_blank()) + 
+    coord_quickmap()
+```
+
+
+::: Challenge
+
+# Challenge: Plot Using Custom Breaks
+
+Create a plot of the TU Delft Digital Surface Model (`DSM_TUD`) that has:
+
+1. Six classified ranges of values (break points) that are evenly divided among the range of pixel values.
+2. Axis labels.
+3. A plot title.
+
+::: solution
```

Comment:
The solution starts on top of the '3. A plot title' line for me
<img width="445" alt="Capture" src="https://github.com/carpentries-incubator/r-geospatial-urban/assets/63798892/0601f4bc-5b29-499b-9886-d48e2d6d79e3">
.

Reply:
Thanks for pointing this out, @Selkubi. The issue was that the callout label was `Challenge`, upper case, instead of `challenge`. It should work now.